src/gui/TrackContainerView.cpp

```diff
@@ -156,44 +156,45 @@ void TrackContainerView::removeTrackView( TrackView * _tv )
 
 
 
-void TrackContainerView::moveTrackViewUp( TrackView * _tv )
+void TrackContainerView::moveTrackView( TrackView * trackView, int indexTo )
 {
-	for( int i = 1; i < m_trackViews.size(); ++i )
+	int indexFrom = m_trackViews.indexOf( trackView );
+
+	BBTrack::swapBBTracks( trackView->getTrack(),
+			m_trackViews[indexTo]->getTrack() );
+
+	m_scrollLayout->removeWidget( trackView );
+	m_scrollLayout->insertWidget( indexTo, trackView );
+
+	qSwap( m_tc->m_tracks[indexTo], m_tc->m_tracks[indexFrom] );
+	m_trackViews.swap( indexTo, indexFrom );
+
+	realignTracks();
+}
+
+
+
+
+void TrackContainerView::moveTrackViewUp( TrackView * trackView )
+{
+	int index = m_trackViews.indexOf( trackView );
+
+	if ( index > 0 )
 	{
-		TrackView * t = m_trackViews[i];
-		if( t == _tv )
-		{
-			BBTrack::swapBBTracks( t->getTrack(),
-					m_trackViews[i - 1]->getTrack() );
-			m_scrollLayout->removeWidget( t );
-			m_scrollLayout->insertWidget( i - 1, t );
-			qSwap( m_tc->m_tracks[i-1], m_tc->m_tracks[i] );
-			m_trackViews.swap( i - 1, i );
-			realignTracks();
-			break;
-		}
+		moveTrackView( trackView, index - 1 );
 	}
 }
 
 
 
 
-void TrackContainerView::moveTrackViewDown( TrackView * _tv )
+void TrackContainerView::moveTrackViewDown( TrackView * trackView )
 {
-	for( int i = 0; i < m_trackViews.size()-1; ++i )
+	int index = m_trackViews.indexOf( trackView );
+
+	if ( index < m_trackViews.size() - 1 )
```

Comment:
Yeah, we can, but this code will probably not even be used until we figure this out.